### PR TITLE
DS3: Use slot_data instead of the external Json file 

### PR DESCRIPTION
**Pull request by Marechal-L**
## What is this fixing or adding?
It removes the Json file declared in downloads.py and uploads.py and uses the fill_slot_data method instead.

## How was this tested?
Tested with the new client to make sure that parsing the seed data at runtime instead of at loading was not a problem

## If this makes graphical changes, please attach screenshots.
It changes some words of a sentence and adds a new parameter to the joining explanation
image

### CODE REVIEW

File: worlds/dark_souls_3/__init__.py [hunk 1]
```diff
@@ -287,7 +290,7 @@ def generate_output(self, output_directory: str):
...
         }
 
-        # generate the file
-        filename = f"AP-{self.world.seed_name}-P{self.player}-{self.world.player_name[self.player]}.json"
-        with open(os.path.join(output_directory, filename), 'w') as outfile:
-            json.dump(data, outfile)
+        return slot_data
+
+    def generate_output(self, output_directory: str):
+        pass
```
Comment (black-sliver) [inline worlds/dark_souls_3/__init__.py:296]: you don't have to override it if it's empty
```suggestion
```

Comment (Berserker66) [reply worlds/dark_souls_3/__init__.py:296]: In fact you really shouldn't as the empty override spawns a thread worker

**Review by black-sliver — CHANGES_REQUESTED** [review]
Can you split the PR into two
1. generate slot data, don't generate json - would merge that for 0.3.6
2. remove upload/download - would merge that for 0.3.7
 
The reason is that by removing the downloads from the webhost, people might not be able to finish their existing games.

**Comment by Marechal-L** [comment]
Yes sure, I'll make the change and split it as suggested

**Review by black-sliver — CHANGES_REQUESTED** [review]
you'll have to change macros.html to 
`                        {% elif patch.game == "Dark Souls III" and patch.data %}`
to avoid empty downloads for new DS3 seeds

otherwise this looks good

**Review by black-sliver — APPROVED** [review]

**Pull request merged**